### PR TITLE
adds ability to transform payload to the config similarly to official client

### PR DESCRIPTION
**Pull request by jduthon**
** Context **

The official client allows to pass as configuration a transform function, that enables consumers to transform the payload before it is sent (see [docs](https://docs.rollbar.com/docs/javascript#transforming-the-payload))

I allowed myself not to support the ability to pass in an async function as it adds a bit of complexity that I don't believe we require (it would be possible to extend later on if needed).

We want the transform ability to be able to alter stack traces as per [the source maps on many domains concept](https://docs.rollbar.com/docs/source-maps#using-source-maps-on-many-domains).

**Changes:**

- Added transform function to the code
- Added README and CHANGELOG documentation for it
- Added tests (and updated the ones with strict checks on configuration)

### CODE REVIEW

File: README.md
Comment (mangs) [inline README.md:117]: Can you explain the difference between this and `customPayloadFields`? They appear very similar. I think we can use what's already there to accomplish this. `customPayloadFields` overwrites payload values in the `report()` function, so if you pass the custom stack trace data to `customPayloadFields.data.body.telemetry.trace.frames` you should get what you're looking for.

I used these docs to figure that out: https://docs.rollbar.com/reference/create-item

**Review by mangs — COMMENTED** [review]
Thanks for the submission. Please let me know your thoughts on my suggestion.

File: README.md
Comment (mangs) [reply README.md:117]: In fact we're not handling stack traces at all in this library, so we could add that. I could have sworn that was in here but I guess not.

Comment (jduthon) [reply README.md:117]: The problem with `customPayloadFields` is:
1. We can only add to the payload, not transform the existing payload, so when the original payload already has stack traces, we can't modify them
2. A custom payload would not have access to the error that triggered the Rollbar logging, except if we forced users to create their own error wrapper handlers (including uncaught) and leave it up to them to gather stack traces from there, which doesn't seem super maintainable to me

We do have stack traces in the library, code normalizing stack frames [here](https://github.com/babbel/rollbar-client.js/blob/main/src/RollbarClientSubmitter.mts#L49), used [here](https://github.com/babbel/rollbar-client.js/blob/main/src/RollbarClientSubmitter.mts#L177).

Comment (mangs) [reply README.md:117]: OK this makes sense to me now, thanks for the explanation. I knew stack frame support was in there somewhere!

One change that would be nice for consistency's sake would be to show the actual default function (like for `setContext`) instead of saying `no-op function`. So in this case

```suggestion
| `transform`        | N        | `Function` | `() => {}`                                                        | Enables users to transform the payload that will be sent to Rollbar. The function must mutate the passed payload. Function signature: `transform(payload, configuration)` where `payload` is the payload object about to be sent to Rollbar and `configuration` is the configuration object                                                                                                                |
```

File: src/RollbarClientSubmitter.mts
Comment (mangs) [inline src/RollbarClientSubmitter.mts:262]: The official client appears to only pass in payload not configuration. It seems like configuration could be removed here without affecting the original intent of this PR. If so, although it deviates somewhat from the official client naming, we could rename this to `transformPayload` which more clearly describes its intent. Thoughts?

Comment (jduthon) [reply src/RollbarClientSubmitter.mts:262]: I had that initially, though then I saw `shouldIgnoreOccurence` (which matches `checkIgnore` from the official library) also receives configuration (despite the official `checkIgnore` doesn't), and I thought it makes sense for those two to have parity. (but I don't mind either way, if you think it's better not to send in the configuration works for me, we don't need it)

On the naming, in theory I'd prefer matching the official version as it makes porting to ours easier... but I can see a lot of things already don't match 1:1 so I can replace as any porting will require a bunch of renaming regardless.

Comment (mangs) [reply src/RollbarClientSubmitter.mts:262]: I think if we combine our ideas we'll end up in a good place. I fear if we leave the name `transform` it will encourage people to use it to modify `this.configuration`. If we rename it to `transformPayload`, it's clear the intent is to do so only for the payload data. It still lets you modify it but it's more clear that's out of scope. I think you including it could be a positive in case someone wants to use the latest configuration data to make a decision about how to transform the payload data, so how about that? `transformPayload(payload.data, this.configuration);`

Comment (jduthon) [reply src/RollbarClientSubmitter.mts]: Done in https://github.com/babbel/rollbar-client.js/pull/16/commits/3adfa4f47dda7522a055e5927dc1ead274649bbc

**Review by mangs — APPROVED** [review]
Looks great!

**Comment by jduthon** [comment]
@mangs I cannot merge the PR, I think because you are listed [as sole code owner](https://github.com/babbel/rollbar-client.js/blob/main/config/github/codeOwners) 😅 
Maybe we should change that to Pineapples. 🤔

**Comment by mangs** [comment]
@jduthon oh I didn't realize that. Sure that's a good idea.

**Comment by mangs** [comment]
I'll merge it in the meantime to unblock you.

**Comment by mangs** [comment]
@jduthon the publish failed because the `package.json` version wasn't bumped. I'm making a new PR now.

**Pull request merged**
